From 7734e9a7abad833a4e0403fd728a4c958853c66c Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 26 Jul 2021 15:35:58 +0300
Subject: [PATCH] Revert "[SW-40084] greco: efuse: read address range fix"

This reverts commit 4844bf3571184f7cf45885bb818126d81e9918a3.

The fix has been moved to btl. hence this is redundant.

Change-Id: I608e5df2395969007fc0abd9416630a923e74270
---

diff --git a/arch/greco/arm/common.c b/arch/greco/arm/common.c
index 36d05ba..8ade6c1 100644
--- a/arch/greco/arm/common.c
+++ b/arch/greco/arm/common.c
@@ -52,7 +52,6 @@
 #include "common.h"
 
 #define FW_WORD_NUM		0x2000	/* FW_SIZE / 2 */
-#define FUSE_READ_RANGE_ALL	0x3F80
 #define PCIE_LTR_DISABLE		0
 #define PCIE_CAP_HDR_NXT_OFFSET		0x1f8
 #define HOST_GPIO_I2C_DEBUG	44
@@ -60,11 +59,6 @@
 u32 get_efuse_read_reg(void)
 {
 	return mmPSOC_EFUSE_READ;
-}
-
-void preboot_efuse_read_range_set(void)
-{
-	regmap_writel(FUSE_READ_RANGE_ALL, mmPSOC_EFUSE_READ_ADDRESS);
 }
 
 bool hl_is_debug_mode(enum hl_debug_hw_func func)
diff --git a/lib/common/efuse.c b/lib/common/efuse.c
index f3552b7..b2088ff 100644
--- a/lib/common/efuse.c
+++ b/lib/common/efuse.c
@@ -28,8 +28,6 @@
 
 static int efuse_init;
 
-__weak void preboot_efuse_read_range_set(void) {}
-
 int preboot_fuse_init(void)
 {
 	int i;
@@ -39,8 +37,6 @@
 		return -1;
 
 	read_reg = get_efuse_read_reg();
-
-	preboot_efuse_read_range_set();
 
 	regmap_writel(1, read_reg);
 
